pages/blogs/[id].tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useState } from "react";
+import { useRouter } from "next/router";
+import blogs from "@/constants/blogDb.json";
+import Divider from "@/components/base/Divider";
+import styles from "@/styles/BlogSingle.module.css";
+import Button from "@/components/base/Button";
+import InfoItem from "@/components/general/InfoItem";
+import { fontSyne, fontSatoshi, fontRobotoMono } from "@/pages/_app";
+
+interface Blog {
+  id: string;
+  title: string;
+  subtitle: string;
+  content: string;
+  image: string;
+  alt: string;
+  info: { date: string; category: string; readingTime: string };
+  author: string;
+  authorInfo: string;
+}
+export default function Blog() {
+  const [currentBlog, setCurrentBlog] = useState<Blog | null>(null);
+  const router = useRouter();
+  const { id } = router.query;
+  console.log(typeof id);
+  useEffect(() => {
+    const blog = blogs.data.filter((blog) => blog.id === id);

```

Comment:
Have a look at JS array find method. The filter is the one works but not the best solution.